zebra-chain/src/transaction.rs

```diff
@@ -335,6 +335,20 @@ impl Transaction {
         }
     }
 
+    /// Returns `effectiveVersion` as described in [§ 7.1 Transaction Encoding and Consensus]:
+    ///
+    /// > `effectiveVersion` [...] is equal to `min(2, version)` when `fOverwintered = 0` and to
+    /// > `version` otherwise.
+    ///
+    /// [§ 7.1 Transaction Encoding and Consensus]: <https://zips.z.cash/protocol/protocol.pdf#txnencoding>
+    pub fn effective_version(&self) -> u32 {
+        if self.is_overwintered() {
+            self.version()
+        } else {
+            std::cmp::min(2, self.version())
+        }
+    }
+
```

Comment:
I wrote this so that the implementation aligns with the logic specified by the protocol (since the consensus rule mentions `effectiveVersion` instead of `version`), but it's superfluous. I'll remove it.

Reply:
Removed in [3e7b308](https://github.com/ZcashFoundation/zebra/pull/9063/commits/3e7b308b159d6b9f913a12c855d017a2742b9694).